### PR TITLE
Remove markdown linting warnings - see #117

### LINKED ISSUE #117: Lines in markdown files are too long, discuss.md is missing
The automatic checks on pull requests produced warnings on a missing file and the length of lines in several markdown files:

```
Missing required file ./_extras/discuss.md
./README.md: Line(s) too long: 1, 2
./_episodes/01-introduction.md: Line(s) too long: 21, 22, 24, 25, 26, 35, 39, 43, 49, 50, 51, 52
./_episodes/02-working-with-openrefine.md: Line(s) too long: 28, 30, 33, 34, 36, 39, 41, 46, 47, 48, 50, 56, 61, 63, 71, 72, 73, 78, 79, 84, 88, 94, 96, 97, 113, 117, 118, 119, 120, 125, 128, 130, 131, 132, 133, 134, 136, 141, 147, 150, 153, 155, 157, 172, 175, 206, 214, 218, 219, 220, 221, 227, 229, 232, 233
./_episodes/03-filter-sort.md: Line(s) too long: 20, 22, 23, 32, 34, 42, 44, 46, 56, 58, 60, 71, 73, 80, 85, 93, 95, 99, 102, 103, 104, 108, 112
./_episodes/04-numbers.md: Line(s) too long: 19, 21, 23, 27, 31, 32, 38, 42, 43, 44, 48
./_episodes/05-scripts.md: Line(s) too long: 20, 21, 22, 27, 31, 35, 37, 38, 41
./_episodes/06-saving.md: Line(s) too long: 20, 25, 30, 34, 35, 36, 40, 41, 42, 46, 47, 53, 54, 57
./_episodes/07-resources.md: Line(s) too long: 18, 23, 24, 29, 30, 32
./_extras/guide.md: Line(s) too long: 8, 12, 13, 20, 32, 37, 47, 52
./reference.md: Line(s) too long: 7, 12, 15, 18, 24, 27, 30, 33, 42
./setup.md: Line(s) too long: 21, 39, 45, 46, 47, 53, 59, 60, 61, 66, 73
Problems detected but ignored (permissive mode).
```

These are not very important problems, but long lines may indeed make editing the text a bit of a hassle.
The maximum line length set in .editorconfig and bin/lesson_check.py is 100 characters for markdown files.

I'd happily accept PRs that update individual files.

### PR DESCRIPTION
**Pull request by jcohen02**
This PR removes a number of linter warnings relating to line length on markdown files, as detailed in #117.

In addition to the line length issues, a few other Markdown formatting changes have been made to address warnings raised when running [`markdownlint`](https://github.com/DavidAnson/markdownlint) on files locally (e.g. trailing spaces, multiple blank lines, ...).

Opening the PR to check that the changes are indeed resolving the warnings - currently a small group of files have been updated, the remainder will be updated in the coming days and additional commits with further changes pushed to this PR.

Closes #117

(Contributing to Hacktoberfest! - please add hacktoberfest-accepted tag as and when PR is accepted and merged 🎉 )

### CODE REVIEW

**Comment by bencomp** [comment]
Thanks for your contribution, @jcohen02! This looks good so far.

As you mention you intend to add more commits to fix the other files, I'll keep this open for now and approve when the other files are ready. (If you don't have time for more, no worries! I could just remove the "closes issue" link and merge.)
I assume #124 comes from checking your changes, which is very helpful :)

**Comment by jcohen02** [comment]
> Thanks for your contribution, @jcohen02! This looks good so far.
> As you mention you intend to add more commits to fix the other files, I'll keep this open for now and approve when the other files are ready.

No problem @bencomp, happy to help. I think this should now be done, if you could approve the running of the GHA workflow, the previous line length errors are now hopefully resolved.

> I assume https://github.com/datacarpentry/openrefine-socialsci/issues/124 comes from checking your changes, which is very helpful :)

Yes, indeed, I've tried to check the rendered site locally as I've been going along with the changes to see that nothing has broken the formatting in the lesson content. I  _think_ this all looks OK but would welcome someone more familiar with the lesson content giving this a quick check through when reviewing.

File: _episodes/05-scripts.md [hunk 1]
```diff
@@ -15,27 +15,45 @@ keypoints:
...
+quick way to clean all of your related data.
 
 ## Saving your work as a script
 
-1. In the `Undo / Redo` section, click `Extract...`, and select the steps that you want to apply to other datasets by clicking the check boxes.
+1. In the `Undo / Redo` section, click `Extract...`, and select the steps that
+   you want to apply to other datasets by clicking the check boxes.
 
-![History](../fig/history.png)
+   ![History](../fig/history.png)
```
Comment (bencomp) [inline _episodes/05-scripts.md:32]: Great, this seems to fix an issue mentioned in #95 (which was closed already, without providing a fix...)

**Review by bencomp — APPROVED** [review]
Thanks for your work! It renders correctly, even with the whitespace inside the Markdown links.

**Pull request merged**